### PR TITLE
feat: process near payments

### PR DESCRIPTION
**Pull request by yomarion**
## Description of the changes
Process Near payments with the `near-api-js` library.

Breaking changes: `connector` is encapsulated within `connectOptions` for methods `hasSufficientFunds` and `isSolvent`.

### CODE REVIEW

File: packages/payment-processor/src/payment/utils-near.ts [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+      GAS_LIMIT,
+      amount,
+    );
+    return;
+  } catch (e) {
+    throw new Error(`Could not pay Near request. Got ${e.message}`);
+  }
+}
+
+// TODO: factorize with payment-detection
```
Comment (yomarion) [inline packages/payment-processor/src/payment/utils-near.ts:65]: I'm on it, it's a minor change

File: packages/payment-processor/src/payment/utils-near.ts [hunk 2]
```diff
@@ -0,0 +1,68 @@
+import { BigNumber, BigNumberish, ethers } from 'ethers';
+import { Contract } from 'near-api-js';
+import { Near, WalletConnection } from 'near-api-js';
+import { Near as NearPaymentDetection } from '@requestnetwork/payment-detection';
+
+export async function isValidNearAddress(nearNetwork: Near, address: string) {
+  try {
+    await nearNetwork.connection.provider.query(`account/${address}`, '');
```
Comment (benjlevesque) [inline packages/payment-processor/src/payment/utils-near.ts:8]: that seems a very slow way to validate a near address?

File: packages/payment-processor/src/payment/utils-near.ts [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    return true;
+  } catch (e) {
+    console.warn(e.stack);
```
Comment (benjlevesque) [inline packages/payment-processor/src/payment/utils-near.ts:11]: I don't think a validator should log, it's expected to throw

**Review by benjlevesque — APPROVED** [review]

File: packages/payment-processor/src/payment/utils-near.ts (hunk 2 above)
Comment (yomarion) [reply packages/payment-processor/src/payment/utils-near.ts:8]: It's the one Near team pushed so...

**Pull request merged**